thrust/thrust/iterator/strided_iterator.h

```diff
@@ -0,0 +1,64 @@
+// SPDX-FileCopyrightText: Copyright (c) 2025 NVIDIA Corporation
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include <thrust/detail/config.h>
+
+#if defined(_CCCL_IMPLICIT_SYSTEM_HEADER_GCC)
+#  pragma GCC system_header
+#elif defined(_CCCL_IMPLICIT_SYSTEM_HEADER_CLANG)
+#  pragma clang system_header
+#elif defined(_CCCL_IMPLICIT_SYSTEM_HEADER_MSVC)
+#  pragma system_header
+#endif // no system header
+
+#include <thrust/iterator/counting_iterator.h>
+#include <thrust/iterator/transform_iterator.h>
+
+THRUST_NAMESPACE_BEGIN
+
+namespace detail
+{
+struct deref
+{
+  template <typename It>
+  _CCCL_HOST_DEVICE auto operator()(It it) const -> it_reference_t<It>
+  {
+    return *it;
+  }
+};
+} // namespace detail
+
+//! \addtogroup iterators
+//! \{
+
+//! \addtogroup fancyiterator Fancy Iterators
+//! \ingroup iterators
+//! \{
+
+template <typename Iterator, typename StrideHolder = detail::empty>
+using strided_iterator =
+  transform_iterator<detail::deref,
+                     counting_iterator<Iterator, use_default, random_access_traversal_tag, use_default, StrideHolder>>;
```

Comment:
While this definition is easy, I wonder whether we should still define a full-blown iterator without using transform and counting iterator, just to have fewer moving parts for the optimizer to fold.

Reply:
Aaaaand we have to create a custom iterator, because of the proxy reference unwrapping in `transform_iterator` :)